source/NuGet.Lucene.Web/Controllers/PackagesController.cs

```diff
@@ -90,12 +90,24 @@ public HttpResponseMessage DownloadPackage(string id, string version="")
                 return result;
             }
 
-            result = Request.CreateResponse(HttpStatusCode.OK);
+            if (Request.Headers.Range != null)
+            {
+                try
+                {
+                    HttpResponseMessage partialResponse = Request.CreateResponse(HttpStatusCode.PartialContent);
+                    partialResponse.Content = new ByteRangeStreamContent(package.GetStream(), Request.Headers.Range, new MediaTypeWithQualityHeaderValue("application/zip"));
+                    return partialResponse;
```

Comment:
Since this returns early, the headers set around line 118 (ETag, Last Modified, Content Disposition) won't be set. Maybe this is okay but I'd prefer consistency.

Have you considered chaining the `if` on line 93 into the `if` on line 108 and setting the `result` var in all code paths instead of returning early?